components/brave_wallet_ui/utils/style.utils.ts

```diff
@@ -26,3 +26,61 @@ export const makePaddingMixin = (defaultPadding: number | string) => css<{
   padding: ${(p) =>
     p?.padding ? sizeCssValue(p.padding) : sizeCssValue(defaultPadding)};
 `
+
+export const getDominantColorFromImageURL = (src: string) => {
+  if (!src) {
+    return undefined
+  }
+
+  // Construct new image from src.
+  let img = new Image()
+  img.src = src
+
+  // Images actual width and height.
+  const { naturalWidth, naturalHeight } = img
+
+  // Higher number will have better precision but may be slower.
+  const precision = 10
+
+  // Calculate canvas sizes.
+  const scaleRatio = Math.min(
+    precision / naturalWidth,
+    precision / naturalHeight
+  )
+  const width = Math.ceil(naturalWidth * scaleRatio)
+  const height = Math.ceil(naturalHeight * scaleRatio)
+  const area = width * height
+
+  // Construct canvas.
+  let context = document.createElement('canvas').getContext('2d')
+
+  // context, width, height and area might be null.
+  if (context && width && height && area) {
```

Comment:
perf: check for valid size before creating the element/context

Reply:
Done!